pkg/controllers/infra/cluster_controller.go

```diff
@@ -141,13 +184,155 @@ func (r *ClusterController) deleteCAPIClusterIfNeeded(ctx context.Context, infra
 }
 
 func (r *ClusterController) reconcile(ctx context.Context, infraCluster *infrav1.Cluster) (ctrl.Result, error) {
-	switch infraCluster.Spec.InfraType {
-	case infrav1.AWSClusterInfraType:
-		// TODO: implement AWS reconcile logic
-		return ctrl.Result{}, nil
-	default:
-		return ctrl.Result{}, errors.Errorf("unsupported infra type %s", infraCluster.Spec.InfraType)
+	// TODO: precheck
+	scope := scope.NewCluster(infraCluster)
+	provider := infraprovider.NewProvider(r.Client, scope)
+	if err := provider.Reconcile(ctx, infraCluster); err != nil {
+		conditions.MarkFalse(infraCluster, infrav1.InfraProviderReadyCondition, infrav1.InfraProviderProvisionFailedReason,
+			capiv1.ConditionSeverityError, err.Error())
+		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile AWS Cluster %s/%s", infraCluster.Namespace, infraCluster.Name)
+	}
+	conditions.MarkTrue(infraCluster, infrav1.InfraProviderReadyCondition)
+	infraCluster.Status.Phase = string(infrav1.ClusterPhaseInfraProvisioned)
```

Comment:
This means every time we reconcile, it is going to set ClusterPhaseInfraProvisioned. If so, not correct